client/request_test.go

```diff
@@ -0,0 +1,72 @@
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"golang.org/x/net/context"
+)
+
+// TestSetHostHeader should set fake host for local communications, set real host
+// for normal communications.
+func TestSetHostHeader(t *testing.T) {
+	testURL := "/test"
+	testCases := []struct {
+		host            string
+		expectedHost    string
+		expectedURLHost string
+	}{
+		{
+			"unix:///var/run/docker.sock",
+			"",
+			"docker",
+		},
+		{
+			"npipe:////./pipe/docker_engine",
+			"",
+			"docker",
+		},
+		{
+			"tcp://0.0.0.0:4243",
+			"",
```

Comment:
I think the issue is with the mock client not the engine-api code itself. The mock client doesn't go thru to same codepath to set the http info.  I modified the test and I think I now see the correct results.
See: https://gist.github.com/duglin/4596113b600bcf44c84edd00b2368aa2
In particular, line 46 followed by setting the transport.

ping @vdemeester to get his opinion and if this is "more correct" then we should probably modify the other testcases to use (aka test) `NewClient()`